assetconnection/opcua/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/assetconnection/opcua/provider/OpcUaOperationProvider.java

```diff
@@ -254,19 +285,25 @@ private Variant getOutputArgument(CallMethodResult methodResult, String name) {
 
     private OperationVariable[] convertResult(CallMethodResult methodResult, Map<String, ElementValue> inoutputParameters, OperationVariable[] inoutputs)
             throws AssetConnectionException {
-        inoutputParameters.entrySet().stream()
-                .filter(x -> hasOutputArgument(x.getKey()))
-                .forEach(LambdaExceptionHelper.rethrowConsumer(x -> ElementValueMapper.setValue(
-                        Stream.of(inoutputs).filter(y -> Objects.equals(x.getKey(), y.getValue().getIdShort())).findAny().get().getValue(),
-                        new PropertyValue(valueConverter.convert(
-                                getOutputArgument(methodResult, x.getKey()),
-                                ((PropertyValue) x.getValue()).getValue().getDataType())))));
-        return Stream.of(outputVariables)
-                .map(LambdaExceptionHelper.rethrowFunction(
-                        x -> hasOutputArgument(x.getValue().getIdShort())
-                                ? convertOutput(getOutputArgument(methodResult, x.getValue().getIdShort()), x)
-                                : null))
-                .toArray(OperationVariable[]::new);
+        for (var param: inoutputParameters.entrySet()) {
+            String idShortMapped = mapOutputIdShortToArgumentName(param.getKey());
+            if (hasOutputArgument(idShortMapped)) {
+                Optional<OperationVariable> ov = Stream.of(inoutputs).filter(y -> Objects.equals(param.getKey(), y.getValue().getIdShort())).findAny();
+                if (ov.isPresent()) {
+                    ElementValueMapper.setValue(ov.get().getValue(), new PropertyValue(valueConverter.convert(
+                            getOutputArgument(methodResult, idShortMapped),
+                            ((PropertyValue) param.getValue()).getValue().getDataType())));
+                }
+            }
+        }
+        List<OperationVariable> list = new ArrayList<>();
```

Comment:
This block is the second intention in this method, returning the filled outputVariables.
Could be rewritten as:
```java
return Stream.of(outputVariables)
        .map(LambdaExceptionHelper.rethrowFunction(
                x -> hasOutputArgument(mapOutputIdShortToArgumentName(x.getValue().getIdShort()))
                        ? convertOutput(getOutputArgument(methodResult, mapOutputIdShortToArgumentName(x.getValue().getIdShort())), x)
                        : null))
        .toArray(OperationVariable[]::new);
```

Reply:
I left the current implementation as it's clearer in my opinion